integration-test/tests/email-autofill.macos.spec.js

```diff
@@ -194,6 +195,80 @@ test.describe('macos', () => {
                 {pixelName: 'autofill_identity', params: {fieldType: 'lastName'}}
             ])
         })
+        test('with an identity only - filling touched select input without label should not autofill select element', async ({page}) => {
+            await forwardConsoleMessages(page)
+            const selectInput = selectInputPage(page)
+
+            await createWebkitMocks()
+                .withAvailableInputTypes(createAvailableInputTypes())
+                .withIdentity(identity)
+                .applyTo(page)
+
+            await applyScript(page)
+
+            const addressCity = 'city2'
+
+            await selectInput.navigate()
+            await selectInput.selectOption(addressCity)
+            await selectInput.selectFirstName(identity.firstName)
+            await page.waitForTimeout(100)
+            await selectInput.assertSelectedValue(addressCity)
+        })
+        test('with an identity only - filling touched select input with label should not autofill select element', async ({page}) => {
+            await forwardConsoleMessages(page)
+            const selectInput = selectInputPage(page)
+
+            await createWebkitMocks()
+                .withAvailableInputTypes(createAvailableInputTypes())
+                .withIdentity(identity)
+                .applyTo(page)
+
+            await applyScript(page)
+
+            const formWithLabel = true
+            const addressCity = 'city2'
+
+            await selectInput.navigate('selectInput', formWithLabel)
+            await selectInput.selectOption(addressCity, formWithLabel)
+            await selectInput.selectFirstName(identity.firstName, formWithLabel)
+            await page.waitForTimeout(100)
+            await selectInput.assertSelectedValue(addressCity, formWithLabel)
+        })
+
+        test('with an identity only - filling touched select input without label should autofill select element', async ({page}) => {
+            await forwardConsoleMessages(page)
+            const selectInput = selectInputPage(page)
+
+            await createWebkitMocks()
+                .withAvailableInputTypes(createAvailableInputTypes())
+                .withIdentity(identity)
+                .applyTo(page)
+
+            await applyScript(page)
+
+            await selectInput.navigate()
+            await selectInput.selectFirstName(identity.firstName)
+            await selectInput.assertSelectedValue(identity.addressCity)
```

Comment:
I think this can suffer from a false positive - if you `page.pause()` after `await selectInput.navigate()`, you'll see the select field already has the value `city1` - this is because the select element in your HTML will default to the first value.

How to solve: either adjust the select element to have a default state of something like 'please select' - or change the mock/identity data.

The logic seems good, it's just that this test will pass regardless :)

Reply:
Damn! Nice catch (and actually discovered a bug because of this). Fixed here: https://github.com/duckduckgo/duckduckgo-autofill/pull/580/commits/384d4ee4a32ce28907d15f07f38d20622cfb4da0. Isolated the select case much better to avoid regression, otherwise the test was skipping select elements where there's a default value 🙈

Also realized the test names were misleading. Fixed that too. Sorry for messing that up!